### PR TITLE
Finalize Round V1 / V2

### PR DESCRIPTION
**Pull request by boudra**
fixes https://github.com/gitcoinco/grants-stack/issues/2981
fixes https://github.com/gitcoinco/grants-stack/issues/2978
fixes https://github.com/gitcoinco/grants-stack/issues/2979

## Description

This one is a bit difficult to QA because of the many steps involved in creating a finalizable round. I have done those steps and recorded a Loom finalizing a round: https://www.loom.com/share/6aaadd27f07846c8bbf1c2e12b6c3592

I tested in local because I had to hack a few things to get some proper matching results with the amount of votes I had.

At the end of the video I noticed that the Fund Grantees page was not working, I found the issue and fixed it:
Screenshot 2024-03-13 at 13 28 46

I also paid out a grantee to make sure the uploaded Merkle root is valid:

Screenshot 2024-03-13 at 13 46 30

I suggest that we leave the QA for Allo v2 when we do the internal E2E round next week, just because the donation flow for Allo v2 is not done yet.

## How to test

- Create a finalizable round (create round, apply, approve, vote, fund, wait for round end)
- Click on Finalize matches
- Verify that payout screen is available
- Verify that finalized matches show

## Checklist

This PR:

- [x] Does it add new payout or donation token? In this case, have they been added to the indexer to avoid it to crash?
- [x] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [x] Doesn't disable eslint rules.
- [x] Doesn't work around the type checker (including but not limited to: type casts, non-null assertions, `@ts-ignore`, unjustified optional values).
- [x] Doesn't contain commented out code.
- [x] If adding/updating a feature, it adds/updates its test script on Notion.

### CODE REVIEW

File: packages/common/src/index.ts [hunk 1]
```diff
@@ -118,10 +118,11 @@ export function fetchProjectPaidInARound(
   roundId: string,
   chainId: ChainId
 ): Promise<Payout[]> {
+  console.log(roundId, chainId);
```
Comment (bhargavaparoksham) [inline packages/common/src/index.ts:121]: leftover console log?

**Review by bhargavaparoksham — APPROVED** [review]
Other than the leftover console log, lgtm!

File: packages/common/src/index.ts (hunk 1 above)
Comment (boudra) [reply packages/common/src/index.ts:121]: whoops

**Review by thelostone-mc — APPROVED** [review]

**Review by michellema1208 — CHANGES_REQUESTED** [review]
@boudra I'm seeing the "estimated matching %" on the Round Stats page is coming up as 0%, even though all projects in my round have received donations from 2 or more unique donors. I remember in the past, usually I would see a nonzero estimate % here - is that right? 

Link: https://manager-staging-git-finalize-round-grants-stack.vercel.app/#/round/0x8a44288efdc9fe593f633fb447fcfb85770c0d57 

Screen Shot 2024-03-13 at 11 28 33 AM

**Comment by boudra** [comment]
> @boudra I'm seeing the "estimated matching %" on the Round Stats page is coming up as 0%, even though all projects in my round have received donations from 2 or more unique donors. I remember in the past, usually I would see a nonzero estimate % here - is that right? I noticed that your round also had a 0% matching distribution on the Round Stats page ... so maybe this is expected?
> 
> Link: https://manager-staging-git-finalize-round-grants-stack.vercel.app/#/round/0x8a44288efdc9fe593f633fb447fcfb85770c0d57
> Screen Shot 2024-03-13 at 11 28 33 AM

The address `0x1fd06f088c720ba3b7a3634a8f021fdd485dca42` has a passport score of 0 and the round has passport enabled, those votes are not counted unfortunately. You should be able to fix it by increasing the score to 15.

**Comment by michellema1208** [comment]
@boudra ohh gotcha - that's my bad! Let me go ahead and continue testing then

**Review by michellema1208 — APPROVED** [review]

**Comment by michellema1208** [comment]
@boudra I approved the PR for now - shared some more context in Discord DMs! We'll definitely make sure to test this flow in our internal testing week next week. 

One question: am I supposed to be able to revise results before the contract has been funded or the round has ended? I uploaded a revised CSV where I gave all of my donations matching, but still seeing 0% matching distribution.

**Pull request merged**